tce/opset/base/base.opp

```diff
@@ -4341,4 +4341,13 @@ Input operand is discarded.</description>
     <in element-count="1" element-width="1" id="1" type="RawData"/>
   </operation>
 
+  <operation>
+    <name>HWLOOP</name>
+    <description>Hardware-loop instruction to iterate given number of instruction. Operand-1: Iteration count and Operand-2: Number of instructions</description>
+    <inputs>2</inputs>
+    <outputs>0</outputs>
+    <in element-count="1" element-width="32" id="1" type="UIntWord"/>
+    <in element-count="1" element-width="32" id="2" type="UIntWord"/>
+  </operation>
```

Comment:
Does it need the flags for having internal state? Also we must ensure the scheduler doesn't move it away from the end of BB.

Reply:
We just set them as control-flow operation; that will force this instruction to be scheduled at the end of BB.

Reply:
Yes, but to be strict, it should also have the state/affects properties, otherwise  there might be illegal code hoisting above/below it in some corner cases. Usually it's a good idea to be more strict than leave things hanging with the compiler related properties. Another question is: Don't we have a "control flow operation" flag in .opp? I thought we did. It would be better to add one instead of monkey patching it at Operation load time.

Reply:
We are allowed to add only one extra property for instruction in the opp (this might sound not right, but that is case. All the instr in base.opp respects this restriction).
I added `control-flow` to allow the BBs to be generated with proper control-flow.
Note that, we set isTerminator flag in TDGen which will ensures that the instruction is the last instruction in the BB.

Reply:
> We are allowed to add only one extra property for instruction in the opp (this might sound not right, but that is case. All the instr in base.opp respects this restriction).

Yes indeed sounds like a bug. Should be easy to fix in the schema, right?

I'm not worried about the LLVM-side issues, but the TCE scheduler. E.g. BB filler could move code around illegally (perhaps).

Reply:
changing the order of "side-effect" and "control-flow" somehow bypasses the issue.

Anyway, now the "side-effect" flag is also added. We can leave the schema bug for another time.